src/prommis/solid_handling/crusher.py

```diff
@@ -0,0 +1,205 @@
+**default** = False. Crush units do not support dynamic behavior.""",
+        ),
+    )
+    CONFIG.declare(
+        "has_holdup",
+        ConfigValue(
+            default=False,
+            domain=In([False]),
+            description="Holdup construction flag - must be False",
+            doc="""Indicates whether holdup terms should be constructed or not.
+**default** - False. Crush units do not have defined volume, thus
+this must be False.""",
+        ),
+    )
+    CONFIG.declare(
+        "material_balance_type",
+        ConfigValue(
+            default=MaterialBalanceType.componentTotal,
+            domain=In(MaterialBalanceType),
+            description="Material balance construction flag",
+            doc="""Indicates what type of mass balance should be constructed,
+**default** - MaterialBalanceType.useDefault.
+**Valid values:** {
+**MaterialBalanceType.useDefault - refer to property package for default
+balance type
+**MaterialBalanceType.none** - exclude material balances,
+**MaterialBalanceType.componentPhase** - use phase component balances,
+**MaterialBalanceType.componentTotal** - use total component balances,
+**MaterialBalanceType.elementTotal** - use total element balances,
+**MaterialBalanceType.total** - use total material balance.}""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package",
+        ConfigValue(
+            default=useDefault,
+            domain=is_physical_parameter_block,
+            description="Property package to use for control volume",
+            doc="""Property parameter object used to define property calculations,
+**default** - useDefault.
+**Valid values:** {
+**useDefault** - use default package from parent model or flowsheet,
+**PropertyParameterObject** - a PropertyParameterBlock object.}""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package_args",
+        ConfigBlock(
+            implicit=True,
+            description="Arguments to use for constructing property packages",
+            doc="""A ConfigBlock with arguments to be passed to a property block(s)
+and used when constructing these,
+**default** - None.
+**Valid values:** {
+see property package for documentation.}""",
+        ),
+    )
+
+    def build(self):
+        """
+        Begin building model (pre-DAE transformation).
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        # Call UnitModel.build to setup dynamics
+        super(CrushAndBreakageUnitData, self).build()
+
+        # Build Control Volume
+        # Build Control Volume
+        self.control_volume = ControlVolume0DBlock(
+            dynamic=False,
+            has_holdup=False,
+            property_package=self.config.property_package,
+            property_package_args=self.config.property_package_args,
+        )
+
+        self.control_volume.add_state_blocks(has_phase_equilibrium=False)
+
+        self.control_volume.add_material_balances(
```

Comment:
I had a version of the code that inlcudes energy and momentum balance and then deleted it (commit name: deleted MomentumbalanceType from import). The reason is that when add energy and momentum balances, there will be AtrributeError during pytest for energy balance: 

> CoalRefuseStateBlockData' object has no attribute 'default_energy_balance_type

'.   for more details of the error: 

> raise PropertyNotSupportedError(
> idaes.core.util.exceptions.PropertyNotSupportedError: fs.unit.control_volume.properties_out[0.0] default_energy_balance_type is not supported by property package (property is not listed in package metadata properties)

for momentum balance will have error:

> raise AttributeError("Unknown attribute '%s'" % name)
> AttributeError: Unknown attribute 'has_pressure_change'
because of these errors, I end up remove those two property balances.
.

Reply:
Those indicate that you are missing things at the unit model level (and maybe the tests). First, you need config arguments to let the user turn these on/off as needed (probably default to off in these cases) and only build the relevant terms if required. You might also need to write some specific test cases (with dummy properties) to test these options as the current property package does not support them.